pkg/operator/resource_reconciler.go

```diff
@@ -336,7 +339,7 @@ func (rr *ResourceReconciler) OnUpdate(old, cur interface{}) {
 		return
 	}
 
-	if rr.DeletionInProgress(mCur) {
+	if !k8sutil.IsStatusCleanupFinalizerPresent(mCur.GetFinalizers()) && rr.DeletionInProgress(mCur) {
```

Comment:
what about updating DeletionInProgress() to check for the presence of the finalizer? It would simplify the change.